Production/src/BaseTupleProducer.cc

```diff
@@ -470,9 +490,44 @@ void BaseTupleProducer::FillGenParticleInfo()
             particles_to_store.push_back(top_bar);
     }
 
-    for(auto particle : particles_to_store) {
+    for(unsigned n = 0; n < particles_to_store.size();++n) {
+        const auto particle = particles_to_store.at(n);
+
+        eventTuple().genParticles_indexes.push_back(n);
         eventTuple().genParticles_pdg.push_back(particle->pdgId());
+        eventTuple().genParticles_status.push_back(particle->status());
         eventTuple().genParticles_p4.push_back(ntuple::LorentzVectorM(particle->p4()));
+
+        const auto first_mother_ptr = dynamic_cast<const reco::GenParticle*>(particle->mother(0));
+        const auto second_mother_ptr = dynamic_cast<const reco::GenParticle*>(particle->mother(1));
+
+        size_t first_mother_index = 0;
```

Comment:
- don't repeat the same code twice: create an array and do extract indices in the loop.
- can be initialized with -1, so will need only "else" section
- use integer to store index